### DISCUSSION TITLE
The `wire:key` again ...

### DISCUSSION THREAD

**Discussion by robsontenorio** [livewire/livewire#9090] [1. Help] [answered]
### Livewire version

3.5.17

### Laravel version

11.35.1

### Which PHP version are you using?

PHP 8.3

### What is the problem?

Why the child component (detail) does not persist the state? Is it a bug or Livewire does not support this case?

What is wrong?

- Have `wire:key` at root loop component.
- Have `wire:key` at livewire component tag

How to reproduce:

1) fill any input of "detail component"
2) Search for "xxx", will not list anything (OK!)
3) Clear "xxxx" and all is listed again, but not persisted values (PROBLEM!)

https://github.com/user-attachments/assets/63b35e8e-d3ac-4a71-9b87-b9ea60cc9937

### Code snippets

**Main page**

```php
<?php

use App\Models\User;
use Livewire\Volt\Component;

new class extends Component {
    
    public string $search = '';

    public function with(): array
    {
        return [
            'users' => User::take(10)->whereLike('name', "%{$this->search}%")->get()
        ];
    }
}; ?>

<div>

    <input wire:model.live="search" class="border mb-5" placeholder="Search name..." />

    <div>
        @foreach($users as $user)
            <div wire:key="root-{{ $user->id }}">
                {{ $user->name }} <br>
                <livewire:detail :id="$user->id" wire:key="user-{{ $user->id }}" />
            </div>
        @endforeach
    </div>
</div>
```

**detail component**
```php
<?php

use Livewire\Volt\Component;

new class extends Component {
    
    public int $id;

    public string $nickname = '';
}; ?>

<div class="pb-3">
    <b>ID:</b> {{ $id }} <br>
    <input wire:model.live="nickname" class="border" /> / {{ $nickname }} <br>
</div>

```

### Wirebox URL

https://wirebox.app/b/y83nq

### How do you expect it to work?

Persist the state of child component.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my help request will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
i've tried this and this is my though:

i think it's not the `wire:key` problem, it will persist when the element is not removed (and it removed because the user with the searched value is not found). it's the basic alpine morph behavior. 

try fill `nickname` input then search user with name `alvera` and then clear `search` input, it will persisted because the component is not removed.

isn't `:key` when looping the child component (or `wire:key` is works too)? https://livewire.laravel.com/docs/nesting#rendering-children-in-a-loop

```blade
<livewire:detail :id="$user->id" :key="$user->id" />
```

**Comment by joshhanley** [comment, marked as answer]
@robsontenorio the issue is that when you filter out the components, they are removed from the page and the components destroyed. Then when you clear the filtering, a new component is created. That's why you're state isn't being retained.

I would suggest doing the filtering on the front end and hide the results you don't need (using like Tailwind `hidden` or `x-show`), that way when the filter is cleared, the component and it's data will still be intact and you can just show it again

**Comment by robsontenorio** [comment]
Thank you guys !
